hikari/api/rest.py

```diff
@@ -1154,27 +1458,215 @@ async def execute_webhook(
+                as an attachment using non-blocking code internally and streamed
+                using bit-inception where possible. This depends on the
+                type of `concurrent.futures.Executor` that is being used for
+                the application (default is a thread pool which supports this
+                behaviour).
+
+        Returns
+        -------
+        hikari.messages.Message
+            The created message.
+
+        Raises
+        ------
+        builtins.ValueError
+            If more than 100 unique objects/entities are passed for
+            `role_mentions` or `user_mentions`.
+        builtins.TypeError
+            If both `attachment` and `attachments` are specified or if both
+            `embed` and `embeds` are specified.
+        hikari.errors.BadRequestError
+            This may be raised in several discrete situations, such as messages
+            being empty with no attachments or embeds; messages with more than
+            2000 characters in them, embeds that exceed one of the many embed
+            limits; too many attachments; attachments that are too large;
+            invalid image URLs in embeds; users in `user_mentions` not being
+            mentioned in the message content; roles in `role_mentions` not
+            being mentioned in the message content.
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.NotFoundError
+            If the channel is not found.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """  # noqa: E501 - Line too long
 
     @abc.abstractmethod
     async def fetch_gateway_url(self) -> str:
-        ...
+        """Fetch the gateway url.
+
+        !!! note
+            This endpoint does not require any valid authorization.
+
+        Raises
+        ------
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
 
     @abc.abstractmethod
     async def fetch_gateway_bot(self) -> sessions.GatewayBot:
-        ...
+        """Fetch the gateway sharding requirements for the bot user.
+
+        !!! note
+            The bot user is determined by the token used in the request.
+            Providing a non-`Bot` token will result in undefined behaviour.
+
+        Returns
+        -------
+        hikari.sessions.GatewayBot
+            The gateway bot.
+
+        Raises
+        ------
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
 
     @abc.abstractmethod
     async def fetch_invite(self, invite: invites.Inviteish) -> invites.Invite:
-        ...
+        """Fetch an existing invite.
```

Comment:
Is it necessary to say that these endpoints all target "existing invites" when this is how the whole rest api works rn and we document the hikari.errors.NotFoundError case which suggests that this only works on existing invites anyway?